azure/compute/src/main/java/org/jclouds/azurecompute/compute/AzureComputeService.java

```diff
@@ -0,0 +1,100 @@
+import com.google.common.base.Predicate;
+import com.google.common.base.Supplier;
+import com.google.common.util.concurrent.ListeningExecutorService;
+
+@Singleton
+public class AzureComputeService extends BaseComputeService {
+   @Inject
+   protected AzureComputeService(ComputeServiceContext context, Map<String, Credentials> credentialStore,
+                                 @Memoized Supplier<Set<? extends Image>> images,
+                                 @Memoized Supplier<Set<? extends Hardware>> hardwareProfiles,
+                                 @Memoized Supplier<Set<? extends Location>> locations, ListNodesStrategy listNodesStrategy,
+                                 GetImageStrategy getImageStrategy, GetNodeMetadataStrategy getNodeMetadataStrategy,
+                                 CreateNodesInGroupThenAddToSet runNodesAndAddToSetStrategy, RebootNodeStrategy rebootNodeStrategy,
+                                 DestroyNodeStrategy destroyNodeStrategy, ResumeNodeStrategy resumeNodeStrategy,
+                                 SuspendNodeStrategy suspendNodeStrategy, Provider<TemplateBuilder> templateBuilderProvider,
+                                 @Named("DEFAULT") Provider<TemplateOptions> templateOptionsProvider,
+                                 @Named(TIMEOUT_NODE_RUNNING) Predicate<AtomicReference<NodeMetadata>> nodeRunning,
+                                 @Named(TIMEOUT_NODE_TERMINATED) Predicate<AtomicReference<NodeMetadata>> nodeTerminated,
+                                 @Named(TIMEOUT_NODE_SUSPENDED) Predicate<AtomicReference<NodeMetadata>> nodeSuspended,
+                                 InitializeRunScriptOnNodeOrPlaceInBadMap.Factory initScriptRunnerFactory, InitAdminAccess initAdminAccess,
+                                 RunScriptOnNode.Factory runScriptOnNodeFactory, PersistNodeCredentials persistNodeCredentials,
+                                 Timeouts timeouts, @Named(Constants.PROPERTY_USER_THREADS) ListeningExecutorService userExecutor,
+                                 Optional<ImageExtension> imageExtension, Optional<SecurityGroupExtension> securityGroupExtension) {
+      super(context, credentialStore, images, hardwareProfiles, locations, listNodesStrategy, getImageStrategy,
+               getNodeMetadataStrategy, runNodesAndAddToSetStrategy, rebootNodeStrategy, destroyNodeStrategy,
+               resumeNodeStrategy, suspendNodeStrategy, templateBuilderProvider, templateOptionsProvider, nodeRunning,
+               nodeTerminated, nodeSuspended, initScriptRunnerFactory, initAdminAccess, runScriptOnNodeFactory,
+               persistNodeCredentials, timeouts, userExecutor, imageExtension, securityGroupExtension);
+   }
+
+   /**
+    * Returns template options, except of type {@link org.jclouds.azurecompute.compute.options.AzureTemplateOptions}.
+    */
+   @Override
+   public AzureTemplateOptions templateOptions() {
+      return AzureTemplateOptions.class.cast(super.templateOptions());
+   }
+}
```

Comment:
There is no need to subclass the BaseComputeService just to override this. No one will directly use this concrete implementation, and the injector will already provide the right instance of the options class, as a binding for it should be configured in the ComputeserviceContextModule.
This class could be removed.